ops/model.py

```diff
@@ -1044,62 +1034,59 @@ class Container:
-        return self._completed
-
-    def is_ready(self) -> '_ContainerReady':
+    def can_connect(self) -> bool:
         """Check whether or not Pebble is ready as a simple property.
 
-        :meth:`is_ready` returns a :class:_ContainerReady `contextmanager` which
-         can be used in charms to wrap :class:`Container` operations which depend
-         on the Pebble backend being available. When `is_ready` is used, exceptions
-         from the underlying Pebble operations will log error messages rather than
-         raising exceptions.
+        :meth:`can_connect` returns a bool that indicates whether the Pebble API is available at
+        the time the method is called. It does not guard against the Pebble API becoming
+        unavailable, and should be treated as a 'point in time' status only.
 
-        Example::
-
-            container = self.unit.get_container("example")
-            with container.is_ready() as c:
-                c.pull('/does/not/exist')
-
-                # This point of execution will not be reached if an exception
-                # was caught earlier
-                c.get_service("foo")
-            c.completed # False
-
-            This will result in an `ERROR` log from PathError, but not a
-            traceback. In addition, the block running inside the contextmanager
-            will exit and return to the previous point of execution. Whether
-            or not the block completed successfully is available as a property
-
-        :meth:`is_ready` can also be used as a bare function, which will log an
-        error if the container is not ready.
+        If the Pebble API later fails, serious consideration should be given as to the reason for
+        this.
 
         Example::
 
-            if container.is_ready():
-                do_something()
+            container = self.unit.get_container("example")
+            if container.can_connect():
+                try:
+                    c.pull('/does/not/exist')
+                except ProtocolError, PathError:
+                    # handle it
             else:
-                do_something_else()
+                event.defer()
         """
-        return _ContainerReady(self)
+        try:
+            # TODO: This call to `get_system_info` should be replaced with a call to a more
+            # appropriate endpoint that has stronger connotations of what constitutes a Pebble
+            # instance that is in fact 'ready'.
+            self._pebble.get_system_info()
+        except pebble.ConnectionError as e:
+            logger.debug("Could not connect to Pebble API: %s", e.message())
+            return False
+        except pebble.APIError as e:
+            # An API error is only raised when the Pebble API returns invalid JSON, or the response
+            # cannot be read. Both of these are a likely indicator that something is wrong.
+            logger.debug("Got an unexpected response from the Pebble API: %s", str(e))
```

Comment:
Isn't this particular one a valid case for `.error` even in the context of this function?

Reply:
I'll upgrade this one to a warning.